packages/protocol/src/action-protocol/client-notification.ts

```diff
@@ -13,53 +13,113 @@
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `ServerStatusActions`.
  */
-export class GLSPServerStatusAction implements Action {
-    static KIND = 'serverStatus';
-
-    constructor(
-        public severity: ServerSeverity,
-        public message: string,
-        public timeout = -1,
-        readonly kind = GLSPServerStatusAction.KIND
-    ) {}
+export interface ServerStatusAction extends Action {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof ServerStatusAction.KIND;
+    /**
+     * The severity of the status.
+     */
+    severity: ServerSeverity;
+
+    /**
+     * The message describing the status.
+     */
+    message: string;
+
+    /**
+     * Timeout after which a displayed status disappears..
+     */
+    timeout?: number;
 }
 
-export function isGLSPServerStatusAction(action: any): action is GLSPServerStatusAction {
-    return (
-        isActionKind(action, GLSPServerStatusAction.KIND) &&
-        isString(action, 'severity') &&
-        isString(action, 'message') &&
-        isNumber(action, 'timeout')
-    );
+export namespace ServerStatusAction {
+    export const KIND = 'serverStatus';
+
+    export function is(object: any): object is ServerStatusAction {
+        return Action.hasKind(object, KIND) && hasStringProp(object, 'severity') && hasStringProp(object, 'message');
+    }
+
+    export function create(message: string, options: { severity?: ServerSeverity; timeout?: number } = {}): ServerStatusAction {
+        return {
+            kind: KIND,
+            severity: 'INFO',
+            message,
+            ...options
+        };
+    }
 }
+
 /**
- * This action is typically sent by the server to notify the user about something of interest.
+ * The possible server status severity levels.
  */
-export class ServerMessageAction implements Action {
-    static KIND = 'serverMessage';
-
-    constructor(
-        public severity: ServerSeverity,
-        public message: string,
-        public details?: string,
-        public timeout = -1,
-        readonly kind = ServerMessageAction.KIND
-    ) {}
+
+export type ServerSeverity = 'NONE' | 'INFO' | 'WARNING' | 'ERROR' | 'FATAL' | 'OK';
+
+/**
+ * This action is sent by the server to notify the user about something of interest. Typically this message is handled by
+ * the client by prompting a message with the application's message service.
+ * If a timeout is given the respective message should disappear after the timeout is reached.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `ServerMessageActions`.
+ */
+export interface ServerMessageAction extends Action {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof ServerMessageAction.KIND;
+
+    /**
+     * The severity of the message.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.